<img src="https://raw.githubusercontent.com/carubbi/MQ/b58af070c45fb17b77640c06c0016859ad366b4d/notebooks/assets/imgs/UNIFOR_logo.png" width="400">
<br>
<b>
<font size="6" face="arial" color="blue">
    Graduação em Ciência da Computação
</font>
</b>
<br>
<b>
<font size="4" face="arial">
    Disciplina: Métodos Quantitativos em Computação
</font>
</b>

**Orientador: Prof. Me. Ricardo Carubbi** <br>
*Docente da Graduação e Pós-Graduação em Ciência de Dados e Inteligência Artificial*<br>
*Laboratório de Ciência de Dados e Inteligência Artificial*<br>
*Universidade de Fortaleza*<br>

# Aula 2 — Fundamentos estatísticos e investigação com dados

## 1. Objetivos de aprendizagem

Ao final da aula, o aluno será capaz de:

- formular uma pergunta estatística compatível com os dados disponíveis;
- distinguir população de interesse, população finita de referência, unidade de análise, unidade observacional e amostra;
- comparar amostras aleatórias e reconhecer a variabilidade amostral;
- avaliar se um procedimento de seleção é adequado ao objetivo da investigação;
- diferenciar variabilidade amostral de viés de seleção;
- distinguir descrição, inferência e afirmação causal;
- delimitar o alcance das conclusões com base na origem e na seleção dos dados.

## 2. Agenda

1. Pergunta estatística e preparação dos dados — 20 min
2. Investigação estatística e variabilidade — 20 min
3. População, amostra, representatividade e vieses — 40 min
4. Descrição, inferência e alcance das conclusões — 20 min

**Código apresentado pronto:** importação do pandas, carregamento do arquivo, seleção das colunas e definição da população finita de referência. As demais operações serão desenvolvidas e interpretadas com a turma.

## 3. Pergunta estatística e preparação dos dados

> O que pode ser descrito com os pinguins observados e até onde essas conclusões podem ser generalizadas?

Uma pergunta estatística antecipa que as respostas podem variar entre as unidades observadas. Ela também precisa ser compatível com as variáveis disponíveis e com a maneira pela qual os dados foram obtidos. Antes de calcular qualquer resultado, é necessário identificar **quem ou o que foi observado**, **qual conjunto se pretende conhecer** e **quais informações foram registradas**.

### 3.1 O que é Estatística e como se investiga

A **Estatística** é a área do conhecimento que reúne métodos para obter, organizar, analisar e interpretar dados, considerando a variabilidade para compreender fenômenos e apoiar decisões. Ela não se limita aos cálculos: relaciona a pergunta, a obtenção dos dados, a análise e a conclusão (BARBETTA; BORNIA; REIS, seções 1.1–1.2; ARAÚJO; SILVA, seções 1.1–1.3).

Uma investigação estatística articula etapas dependentes entre si:

1. delimitar o problema;
2. formular uma pergunta compatível com os dados;
3. definir a população de interesse, a unidade de análise e a unidade observacional;
4. examinar ou planejar a obtenção dos dados;
5. analisar a variabilidade e os padrões;
6. comunicar a conclusão e explicitar seu alcance.

Uma falha nas primeiras etapas não é corrigida automaticamente por cálculos corretos nas etapas finais (ARAÚJO; SILVA, seções 1.1–1.3; BARBETTA; BORNIA; REIS, seções 1.2 e 2.1).

![Etapas de uma investigação estatística, da definição do problema às conclusões.](../assets/imgs/u1_a02/figura_2_1_processo_investigacao.png)

**Figura 1 - Etapas de uma investigação estatística.** A definição do problema orienta a coleta, a análise e a interpretação; por isso, a conclusão precisa responder à pergunta inicial e respeitar as limitações dos dados. Fonte: Barbetta, Bornia e Reis (2010, figura 2.1).

### 3.2 Contexto dos dados

A aplicação utiliza 344 registros do conjunto Palmer Penguins. Cada linha corresponde a um pinguim observado no contexto dos estudos documentados; as colunas registram características como espécie, ilha e massa corporal.

Para esta aula, distinguiremos:

- **população finita de referência:** as 344 linhas disponíveis no arquivo;
- **unidade de análise:** o pinguim registrado, sobre o qual serão formuladas as conclusões;
- **unidade observacional:** o pinguim no qual as características foram observadas e registradas;
- **amostra:** um subconjunto selecionado dessas linhas.

Nesta aplicação, a unidade de análise e a unidade observacional coincidem. Em outras investigações, a informação pode ser coletada em uma unidade e a conclusão formulada sobre outra. Essa convenção permite estudar seleção e variabilidade dentro do arquivo, mas não transforma os registros disponíveis em uma representação automática de todos os pinguins das três espécies (MORDEN, 2024, seção 3.7; HORST; HILL; GORMAN, 2020).

In [1]:
# Importar a biblioteca utilizada na aula.
import pandas as pd


In [2]:
# Definir a localização e carregar os dados.
arquivo_dados = "https://raw.githubusercontent.com/carubbi/MQ/main/data/raw/penguins_raw.csv"
df = pd.read_csv(arquivo_dados)

# Visualizar as primeiras observações.
df.head()


,studyName,Sample Number,Species,Region,Island,Stage,Individual ID,Clutch Completion,Date Egg,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Sex,Delta 15 N (o/oo),Delta 13 C (o/oo),Comments
0,PAL0708,1,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N1A1,Yes,2007-11-11,39.1,18.7,181.0,3750.0,MALE,NaN,NaN,Not enough blood for isotopes.
1,PAL0708,2,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N1A2,Yes,2007-11-11,39.5,17.4,186.0,3800.0,FEMALE,8.94956,-24.69454,NaN
2,PAL0708,3,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N2A1,Yes,2007-11-16,40.3,18.0,195.0,3250.0,FEMALE,8.36821,-25.33302,NaN
3,PAL0708,4,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N2A2,Yes,2007-11-16,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Adult not sampled.
4,PAL0708,5,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N3A1,Yes,2007-11-16,36.7,19.3,193.0,3450.0,FEMALE,8.76651,-25.32426,NaN


**Tabela 1 - Primeiras observações do dataset.** Cada linha registra um pinguim e cada coluna armazena uma característica ou informação do estudo. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

Para a atividade, serão mantidas apenas as colunas necessárias:

- `studyName`: estudo em que a observação foi registrada;
- `Individual ID`: identificador do indivíduo dentro do estudo;
- `Species`: espécie do pinguim;
- `Island`: ilha onde o pinguim foi observado;
- `Body Mass (g)`: massa corporal, em gramas.

Como o identificador pode se repetir entre estudos, a unidade é identificada pelo par `studyName` + `Individual ID` (HORST; HILL; GORMAN, 2020).


In [3]:
# Escolher as colunas necessárias.
colunas_amostras = [
    "studyName",
    "Individual ID",
    "Species",
    "Island",
    "Body Mass (g)",
]

# Criar a população finita de referência.
populacao_referencia = df[colunas_amostras].copy()

# Verificar o número de linhas e colunas.
populacao_referencia.shape


(344, 5)

A saída `(344, 5)` informa que a população finita de referência possui 344 linhas e 5 colunas selecionadas.

### 3.3 Perguntas compatíveis e incompatíveis com os dados

| Pergunta | Avaliação | Justificativa |
| --- | --- | --- |
| Quantos pinguins foram registrados por espécie? | Compatível com os dados | A espécie está registrada no arquivo. |
| Como a massa corporal varia entre os registros válidos? | Compatível com os dados | A massa corporal está registrada para a maioria das observações. |
| A massa de cada pinguim mudou ao longo da vida? | Incompatível com os dados | O arquivo não acompanha repetidamente cada animal ao longo da vida. |
| A ilha causou diferenças de massa corporal? | Incompatível com o desenho do estudo | Os dados são observacionais e não resultam da atribuição experimental de uma ilha. |

A existência de uma coluna permite calcular associações, mas não garante que os dados sustentem conclusões causais ou temporais.

## 4. Investigação estatística e variabilidade

A **variabilidade observada** corresponde às diferenças encontradas entre as unidades registradas: pinguins podem pertencer a espécies distintas, ocupar ilhas diferentes e apresentar massas corporais diferentes. Essas diferenças podem refletir características biológicas, contexto, período, medição ou preparação dos dados.

Um valor incomum não deve ser classificado automaticamente como erro. Primeiro, deve-se verificar a unidade de medida, o procedimento de coleta, o registro original e a coerência com as demais variáveis (BARBETTA; BORNIA; REIS, seção 1.2; MONTGOMERY; RUNGER, seção 1.1.1).

### 4.1 Reconhecer a composição e a variabilidade dos registros

Antes de selecionar amostras, examinaremos a composição da população finita de referência e a variação da massa corporal. Esses resultados servirão de comparação para as seleções posteriores.

In [4]:
# Contar quantas observações existem de cada espécie.
composicao_referencia = populacao_referencia["Species"].value_counts()
composicao_referencia


Species
Adelie Penguin (Pygoscelis adeliae)          152
Gentoo penguin (Pygoscelis papua)            124
Chinstrap penguin (Pygoscelis antarctica)     68
Name: count, dtype: int64

**Tabela 2 - Composição da população de referência por espécie.** A tabela será o parâmetro de comparação para as amostras produzidas nesta aula. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

In [5]:
# Contar quantas observações existem em cada ilha.
composicao_ilhas = populacao_referencia["Island"].value_counts()
composicao_ilhas


Island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

**Tabela 3 - Composição da população de referência por ilha.** As quantidades mostram que as ilhas não aparecem com a mesma frequência no arquivo. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

Antes de sortear, a equipe conhece a composição do conjunto disponível. Isso permite verificar depois quais grupos apareceram, desapareceram ou ficaram super-representados nas amostras.

In [6]:
# Resumir a massa corporal para reconhecer sua variabilidade.
resumo_massa = populacao_referencia["Body Mass (g)"].describe()
resumo_massa


count     342.000000
mean     4201.754386
std       801.954536
min      2700.000000
25%      3550.000000
50%      4050.000000
75%      4750.000000
max      6300.000000
Name: Body Mass (g), dtype: float64

**Tabela 4 - Resumo da massa corporal.** A saída descreve os valores disponíveis de `Body Mass (g)` e permite reconhecer quantidade válida, posição e variabilidade. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

**Registro do estudante:** formule uma pergunta compatível com os dados sobre a massa corporal e explique por que um valor incomum não deve ser classificado automaticamente como erro.

**Resposta:** uma pergunta compatível com os dados é: “Como varia a massa corporal entre os pinguins registrados no arquivo?”. Há 342 valores válidos, entre 2.700 g e 6.300 g. Um valor incomum pode refletir variação biológica legítima; antes de classificá-lo como erro, é necessário verificar a unidade, o procedimento de medição, o registro original e sua coerência com as demais variáveis.

### 4.2 Variabilidade entre amostras

A **variabilidade amostral** aparece quando amostras diferentes, obtidas pelo mesmo procedimento, produzem composições ou estatísticas diferentes. Assim, duas amostras aleatórias de dez registros podem apresentar proporções distintas de espécies e médias distintas de massa corporal.

Essa diferença é esperada em amostras pequenas e não comprova, isoladamente, a existência de viés. O notebook verificará essa ideia comparando duas seleções reproduzíveis da mesma população finita de referência.

## 5. População, amostra, representatividade e vieses

- **População de interesse:** conjunto de unidades sobre o qual se pretende responder.
- **População finita de referência:** conjunto operacional adotado para a comparação interna desta aula.
- **Quadro amostral:** representação das unidades que poderiam ser selecionadas pelo procedimento.
- **Amostra:** subconjunto efetivamente selecionado e analisado.
- **Amostragem:** procedimento usado para selecionar as unidades.
- **Amostragem aleatória simples:** procedimento no qual cada amostra possível de mesmo tamanho tem a mesma probabilidade de ser selecionada.

A população não é uma propriedade fixa do arquivo: ela depende da pergunta. Se o objetivo é descrever somente os registros Gentoo, o subconjunto Gentoo pode ser a população de interesse. Se o objetivo é comparar as três espécies, o mesmo filtro exclui grupos necessários à investigação. A seleção aleatória simples só oferece cobertura adequada quando o quadro amostral inclui as unidades elegíveis; neste notebook, o sorteio é aleatório apenas em relação às linhas disponíveis no arquivo (BARBETTA; BORNIA; REIS, seções 1.6 e 2.2.1; NAVIDI, seção 1.1; PINHEIRO et al., seção 1.2).

![Relação entre população, amostragem, amostra e inferência.](../assets/imgs/u1_a02/figura_2_2_populacao_amostra_inferencia.png)

**Figura 2 - Relação entre população, amostra, amostragem e inferência.** A inferência parte da amostra para responder sobre uma população definida, mas sua validade depende do procedimento de seleção e das condições assumidas. Fonte: Barbetta, Bornia e Reis (2010, figura 2.2).

### 5.1 Representatividade não é sinônimo de tamanho

Uma amostra é útil quando o procedimento de seleção fornece cobertura adequada das unidades relevantes para a pergunta. Aumentar o número de observações pode reduzir a variabilidade amostral, mas não corrige automaticamente a exclusão sistemática de grupos.

Além disso, um sorteio aleatório entre as linhas disponíveis distribui a chance **dentro do arquivo**; ele não corrige limitações da coleta que produziu o próprio arquivo. Portanto, `random_state` torna o sorteio reproduzível, mas não torna a amostra representativa de uma população biológica mais ampla (BARBETTA; BORNIA; REIS, 2010, seção 2.2.1; NAVIDI, 2024, seção 1.1).

### 5.2 Variabilidade e viés não são a mesma coisa

| Fenômeno | Característica | Exemplo nesta aula |
| --- | --- | --- |
| Variabilidade amostral | Resultados mudam entre amostras obtidas pelo mesmo procedimento. | Duas amostras aleatórias de dez registros apresentam composições diferentes. |
| Viés de seleção | O procedimento favorece ou exclui sistematicamente unidades relevantes. | Filtrar somente Gentoo quando o objetivo é comparar as três espécies. |
| Viés de medição | O registro se afasta sistematicamente da característica que se pretende medir. | Deve ser considerado conceitualmente, mas não pode ser diagnosticado apenas com as saídas usadas nesta aula. |

A aleatoriedade aplicada depois de um filtro inadequado não recupera os grupos já excluídos. Por outro lado, diferenças ocasionais entre duas amostras aleatórias não bastam para concluir que houve viés.

### 5.3 Comparar amostras aleatórias reproduzíveis

Duas amostras de mesmo tamanho serão obtidas pelo mesmo procedimento, com sementes diferentes. A comparação permitirá separar diferenças esperadas entre sorteios de um mecanismo sistemático de exclusão.

In [7]:
# Sortear 10 observações sem reposição.
amostra_a = populacao_referencia.sample(
    n=10,
    random_state=42,
)

amostra_a


,studyName,Individual ID,Species,Island,Body Mass (g)
194,PAL0809,N7A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
157,PAL0708,N33A2,Gentoo penguin (Pygoscelis papua),Biscoe,4550.0
225,PAL0809,N55A2,Gentoo penguin (Pygoscelis papua),Biscoe,5200.0
208,PAL0809,N16A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
318,PAL0809,N75A1,Chinstrap penguin (Pygoscelis antarctica),Dream,3550.0
329,PAL0910,N92A2,Chinstrap penguin (Pygoscelis antarctica),Dream,4050.0
319,PAL0809,N75A2,Chinstrap penguin (Pygoscelis antarctica),Dream,3500.0
260,PAL0910,N29A1,Gentoo penguin (Pygoscelis papua),Biscoe,4575.0
114,PAL0910,N61A1,Adelie Penguin (Pygoscelis adeliae),Biscoe,3900.0
220,PAL0809,N53A1,Gentoo penguin (Pygoscelis papua),Biscoe,4700.0


**Tabela 5 - Amostra aleatória A.** Foram sorteadas 10 linhas da população finita de referência. O valor de `random_state` permite reproduzir o mesmo resultado (PANDAS DEVELOPMENT TEAM, 2026). Fonte dos dados: Horst, Hill e Gorman (2020).

In [8]:
# Repetir o sorteio com outra semente.
amostra_b = populacao_referencia.sample(
    n=10,
    random_state=7,
)

amostra_b


,studyName,Individual ID,Species,Island,Body Mass (g)
173,PAL0708,N41A2,Gentoo penguin (Pygoscelis papua),Biscoe,5000.0
30,PAL0708,N21A1,Adelie Penguin (Pygoscelis adeliae),Dream,3250.0
115,PAL0910,N61A2,Adelie Penguin (Pygoscelis adeliae),Biscoe,4075.0
341,PAL0910,N99A2,Chinstrap penguin (Pygoscelis antarctica),Dream,3775.0
141,PAL0910,N80A2,Adelie Penguin (Pygoscelis adeliae),Dream,3475.0
73,PAL0809,N35A2,Adelie Penguin (Pygoscelis adeliae),Torgersen,4150.0
336,PAL0910,N96A1,Chinstrap penguin (Pygoscelis antarctica),Dream,3950.0
328,PAL0910,N92A1,Chinstrap penguin (Pygoscelis antarctica),Dream,3600.0
208,PAL0809,N16A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
28,PAL0708,N18A1,Adelie Penguin (Pygoscelis adeliae),Biscoe,3150.0


**Tabela 6 - Amostra aleatória B.** A mudança de `random_state` produz outra seleção reproduzível de 10 linhas. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

In [9]:
# Contar as espécies presentes nas duas amostras.
composicao_amostra_a = amostra_a["Species"].value_counts()
composicao_amostra_b = amostra_b["Species"].value_counts()

# Reunir as contagens para facilitar a comparação.
comparacao_composicoes = pd.DataFrame(
    {
        "População de referência": composicao_referencia,
        "Amostra A": composicao_amostra_a,
        "Amostra B": composicao_amostra_b,
    }
).fillna(0).astype(int)

comparacao_composicoes


,População de referência,Amostra A,Amostra B
Species,,,
Adelie Penguin (Pygoscelis adeliae),152,1,5
Chinstrap penguin (Pygoscelis antarctica),68,3,3
Gentoo penguin (Pygoscelis papua),124,6,2


**Tabela 7 - Contagens por espécie na população e nas amostras.** As duas amostras foram obtidas pelo mesmo procedimento, mas apresentam contagens diferentes. Como a população possui 344 observações e cada amostra possui 10, a comparação proporcional será feita na próxima tabela. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).


In [10]:
# Calcular as proporções de cada espécie, em porcentagem.
proporcoes_referencia = composicao_referencia / len(populacao_referencia) * 100
proporcoes_amostra_a = composicao_amostra_a / len(amostra_a) * 100
proporcoes_amostra_b = composicao_amostra_b / len(amostra_b) * 100

# Reunir e arredondar as porcentagens.
comparacao_proporcoes = pd.DataFrame(
    {
        "População de referência (%)": proporcoes_referencia,
        "Amostra A (%)": proporcoes_amostra_a,
        "Amostra B (%)": proporcoes_amostra_b,
    }
).fillna(0).round(1)

comparacao_proporcoes


,População de referência (%),Amostra A (%),Amostra B (%)
Species,,,
Adelie Penguin (Pygoscelis adeliae),44.2,10.0,50.0
Chinstrap penguin (Pygoscelis antarctica),19.8,30.0,30.0
Gentoo penguin (Pygoscelis papua),36.0,60.0,20.0


**Tabela 8 - Proporções das espécies na população e nas amostras.** Adelie corresponde a 44,2% da população de referência, mas representa 10,0% da amostra A e 50,0% da amostra B. Gentoo passa de 36,0% na população para 60,0% na amostra A e 20,0% na amostra B. Essas diferenças ilustram a variabilidade da composição entre amostras aleatórias pequenas. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

**Registro do estudante:** a seleção aleatória produziu uma cópia exata da composição da população? Essas diferenças, isoladamente, comprovam viés?

**Resposta:** não. Nenhuma das amostras reproduziu exatamente a composição da população de referência. As diferenças observadas entre sorteios de dez registros são compatíveis com variabilidade amostral e, isoladamente, não comprovam viés. Para caracterizar viés, seria necessário identificar um mecanismo de seleção que favoreça ou exclua sistematicamente grupos relevantes ao objetivo.


In [11]:
# Calcular a média da massa corporal em cada conjunto.
media_massa_referencia = populacao_referencia["Body Mass (g)"].mean()
media_massa_amostra_a = amostra_a["Body Mass (g)"].mean()
media_massa_amostra_b = amostra_b["Body Mass (g)"].mean()

# Reunir e arredondar as médias.
comparacao_medias = pd.Series(
    {
        "População de referência": media_massa_referencia,
        "Amostra A": media_massa_amostra_a,
        "Amostra B": media_massa_amostra_b,
    },
    name="Média da massa corporal (g)",
).round(1)

comparacao_medias


População de referência    4201.8
Amostra A                  4262.5
Amostra B                  3872.5
Name: Média da massa corporal (g), dtype: float64

**Tabela 9 - Média da massa corporal na população e nas amostras.** A média é 4.201,8 g na população de referência, 4.262,5 g na amostra A e 3.872,5 g na amostra B. A mesma estatística assume valores diferentes porque foi calculada em seleções diferentes. Isso evidencia variabilidade amostral e não demonstra, por si só, a existência de viés. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).


### 5.4 O objetivo determina a adequação do filtro

Considere dois objetivos:

- **Objetivo A:** comparar Adelie, Chinstrap e Gentoo. Um filtro que mantém somente Gentoo é inadequado porque impede a comparação pretendida.
- **Objetivo B:** descrever somente os registros Gentoo. O mesmo filtro passa a ser coerente, pois redefine corretamente a população de interesse.

O código não é adequado ou inadequado isoladamente. Sua validade depende da correspondência entre pergunta, população e procedimento de seleção.

#### Objetivo A: descrever as três espécies

Suponha que a equipe queira descrever conjuntamente Adelie, Chinstrap e Gentoo, mas filtre apenas Gentoo antes do sorteio. Primeiro, o filtro será aplicado; depois, o subconjunto será amostrado.

In [12]:
# Criar uma condição para selecionar somente Gentoo.
filtro_gentoo = (
    populacao_referencia["Species"]
    == "Gentoo penguin (Pygoscelis papua)"
)

# Aplicar o filtro.
populacao_gentoo = populacao_referencia[filtro_gentoo]

# Verificar o tamanho do subconjunto.
populacao_gentoo.shape


(124, 5)

In [13]:
# Sortear 10 observações dentro do subconjunto Gentoo.
amostra_gentoo = populacao_gentoo.sample(
    n=10,
    random_state=42,
)

amostra_gentoo


,studyName,Individual ID,Species,Island,Body Mass (g)
170,PAL0708,N40A1,Gentoo penguin (Pygoscelis papua),Biscoe,4800.0
194,PAL0809,N7A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
188,PAL0809,N4A1,Gentoo penguin (Pygoscelis papua),Biscoe,4950.0
228,PAL0809,N58A1,Gentoo penguin (Pygoscelis papua),Biscoe,4600.0
205,PAL0809,N14A2,Gentoo penguin (Pygoscelis papua),Biscoe,5000.0
242,PAL0910,N15A1,Gentoo penguin (Pygoscelis papua),Biscoe,4950.0
208,PAL0809,N16A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
274,PAL0910,N43A1,Gentoo penguin (Pygoscelis papua),Biscoe,5200.0
273,PAL0910,N39A2,Gentoo penguin (Pygoscelis papua),Biscoe,5750.0
156,PAL0708,N33A1,Gentoo penguin (Pygoscelis papua),Biscoe,5400.0


**Tabela 10 - Amostra obtida após o filtro por espécie.** O sorteio é aleatório dentro do subconjunto Gentoo, mas Adelie e Chinstrap não podem ser selecionadas. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

In [14]:
# Verificar a relação observada entre espécie e ilha.
especies_por_ilha = pd.crosstab(
    populacao_referencia["Species"],
    populacao_referencia["Island"],
)

especies_por_ilha


Island,Biscoe,Dream,Torgersen
Species,,,
Adelie Penguin (Pygoscelis adeliae),44,56,52
Chinstrap penguin (Pygoscelis antarctica),0,68,0
Gentoo penguin (Pygoscelis papua),124,0,0


**Tabela 11 - Espécies observadas por ilha.** No dataset, Gentoo aparece somente em Biscoe; Chinstrap, somente em Dream; e Adelie, nas três ilhas. Assim, filtrar uma ilha também pode alterar quais espécies podem integrar a seleção. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

Para o **Objetivo A**, o procedimento que produziu a amostra Gentoo apresenta viés de seleção: o procedimento exclui grupos que pertencem à população definida pelo objetivo. A aleatoriedade aplicada depois do filtro não corrige essa exclusão (BARBETTA; BORNIA; REIS, 2010, seção 2.2.1; NAVIDI, 2024, seção 1.1).


#### Objetivo B: descrever somente Gentoo

Se a pergunta mudar para “como são os pinguins Gentoo registrados no dataset?”, o filtro deixa de ser um erro. A população de interesse passa a ser o subconjunto Gentoo, e a amostra é sorteada dentro do grupo adequado.

O mesmo código pode ser coerente ou inadequado conforme o objetivo da investigação. Por isso, a população de interesse deve ser definida **antes** da seleção dos dados.


### 5.5 Aplicação orientada

Considere a seguinte solicitação:

> A equipe deseja comparar as três espécies, mas decidiu coletar uma amostra apenas na ilha Biscoe.

Execute a próxima célula e registre, em uma célula Markdown:

1. Qual é a população de interesse declarada?
2. Quais espécies podem aparecer na seleção feita em Biscoe?
3. Qual espécie é excluída e qual fica restrita?
4. A amostra permite comparar as três espécies? Justifique com a saída produzida.


In [15]:
# Selecionar as observações da ilha Biscoe.
filtro_biscoe = populacao_referencia["Island"] == "Biscoe"
populacao_biscoe = populacao_referencia[filtro_biscoe]

# Contar as espécies disponíveis após o filtro.
especies_em_biscoe = populacao_biscoe["Species"].value_counts()
especies_em_biscoe


Species
Gentoo penguin (Pygoscelis papua)      124
Adelie Penguin (Pygoscelis adeliae)     44
Name: count, dtype: int64

**Tabela 12 - Espécies disponíveis após o filtro por Biscoe.** A saída fornece a evidência necessária para avaliar se o procedimento atende ao objetivo de comparar as três espécies. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

**Resposta:** a população de interesse declarada reúne as três espécies registradas no arquivo. Após o filtro por Biscoe, somente Gentoo e Adelie podem aparecer: Chinstrap é excluída e Adelie fica restrita às 44 observações dessa ilha. Portanto, a seleção não permite comparar adequadamente as três espécies e apresenta viés em relação ao objetivo declarado.


## 6. Descrição, inferência e alcance das conclusões

| Tipo de afirmação | O que faz | Exemplo |
| --- | --- | --- |
| Descritiva | Resume as observações disponíveis. | Informar quantos registros de cada espécie aparecem no arquivo. |
| Inferencial | Usa uma amostra e um modelo para responder sobre uma população definida. | Estimar a proporção de Adelie em uma população-alvo a partir de uma amostra representativa. |
| Causal | Atribui uma mudança a uma intervenção ou exposição. | Afirmar que a ilha causou determinada massa corporal. |

Os cálculos deste notebook descrevem corretamente o arquivo e as amostras sorteadas dentro dele. Eles não sustentam automaticamente inferências sobre todos os pinguins das espécies nem afirmações causais sobre ilha, espécie ou massa corporal.

### 6.1 Parâmetro, estimador e estimativa

Um **parâmetro** $\theta$ descreve uma característica da população e, em geral, é desconhecido. Uma **estatística amostral** é uma função dos valores da amostra. Quando essa estatística é usada para estimar o parâmetro, a regra definida antes da observação dos dados é o **estimador**:

$$
\widehat{\theta}=T(X_1,X_2,\ldots,X_n).
$$

Depois que os valores são observados, obtém-se a **estimativa**, isto é, o resultado numérico do estimador:

$$
\hat{\theta}=T(x_1,x_2,\ldots,x_n).
$$

Aqui, $X_i$ representa a variável aleatória associada à unidade $i$, $x_i$ representa o valor observado e $n$ é o tamanho da amostra.

A correção do cálculo não garante a validade da generalização. Também são necessários uma população bem definida, um procedimento de seleção justificável e hipóteses compatíveis com o método inferencial (ARAÚJO; SILVA, seção 1.1; PINHEIRO et al., seção 1.1).

### 6.2 Como declarar o alcance

Uma conclusão responsável informa:

- quais observações foram analisadas;
- qual população foi definida;
- como as unidades foram selecionadas;
- quais grupos ou períodos ficaram fora da cobertura;
- se o resultado é descritivo, inferencial ou causal;
- quais limitações impedem uma generalização mais ampla.

No Palmer Penguins, uma formulação apropriada começa por “nos registros disponíveis” ou “na amostra selecionada”. A expressão “em todos os pinguins dessas espécies” exigiria justificativas que o arquivo, isoladamente, não fornece.

In [16]:
# Calcular as contagens e as proporções observadas de cada espécie.
contagens_especies = populacao_referencia["Species"].value_counts()
proporcoes_especies = (contagens_especies / len(populacao_referencia) * 100).round(1)

# Calcular a quantidade válida, o menor e o maior valor de massa corporal.
massa_corporal = populacao_referencia["Body Mass (g)"]
amplitude_massa = pd.Series(
    {
        "Quantidade válida": massa_corporal.count(),
        "Mínimo": massa_corporal.min(),
        "Máximo": massa_corporal.max(),
    }
)

# Reunir as descrições em uma única tabela.
descricao_arquivo = pd.concat(
    [contagens_especies, proporcoes_especies, amplitude_massa],
    keys=["Contagem por espécie", "Proporção por espécie (%)", "Massa corporal (g)"],
    names=["Medida", "Categoria"],
).to_frame(name="Valor")

descricao_arquivo


Valor
Medida                    Categoria                                        
Contagem por espécie      Adelie Penguin (Pygoscelis adeliae)         152.0
                          Gentoo penguin (Pygoscelis papua)           124.0
                          Chinstrap penguin (Pygoscelis antarctica)    68.0
Proporção por espécie (%) Adelie Penguin (Pygoscelis adeliae)          44.2
                          Gentoo penguin (Pygoscelis papua)            36.0
                          Chinstrap penguin (Pygoscelis antarctica)    19.8
Massa corporal (g)        Quantidade válida                           342.0
                          Mínimo                                     2700.0
                          Máximo                                     6300.0

**Tabela 13 - Descrição do arquivo por espécie e massa corporal.** As contagens, proporções e amplitudes pertencem às observações disponíveis. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

**Registro do estudante:** escreva uma conclusão descritiva sustentada pela saída e uma generalização que a seleção disponível não permite sustentar.

**Resposta:** no arquivo, Adelie possui 152 registros (44,2%), Gentoo 124 (36,0%) e Chinstrap 68 (19,8%). Entre os 342 valores válidos de massa corporal, os valores variam de 2.700 g a 6.300 g. Esses resultados não permitem afirmar que as mesmas proporções ou a mesma amplitude caracterizam todos os pinguins dessas espécies, pois o arquivo não é uma amostra probabilística dessa população biológica.

## 7. Síntese

- A pergunta estatística orienta a definição da população e a escolha dos dados.
- Variação entre unidades e variação entre amostras são esperadas.
- Seleção aleatória dentro de um arquivo não elimina limitações da coleta original.
- Representatividade depende do procedimento e da cobertura, não apenas do tamanho.
- Variabilidade amostral não é sinônimo de viés.
- Um filtro deve ser avaliado em relação ao objetivo da investigação.
- Descrever uma base não equivale a generalizar para uma população mais ampla.
- Afirmações causais exigem desenho e evidências compatíveis.

## 8. Estudo e exercícios

### Materiais didáticos

- **Apostila de Métodos Quantitativos:** seções 1.1–1.3, páginas 8–10.
- **Barbetta, Bornia e Reis:** seções 1.2, 1.6 e 2.1–2.2.1, páginas 13 e 18–31.
- **Palmer Penguins:** documentação da fonte, das variáveis e da coleta.

### Exercícios indicados

- Banco de questões e provas 2026.2, questões 5 e 6, página 8: investigação estatística, população e amostra.
- Barbetta, Bornia e Reis, exercício 1.2, página 23: população, amostra e amostragem.
- Barbetta, Bornia e Reis, exercício 2.7, página 34: amostragem e representatividade.

## 9. Referências

ARAÚJO, Cledinaldo Castro; SILVA, Vera Lúcia. **Métodos quantitativos para engenharia: questões contextualizadas & exercícios**. 2. ed.

BARBETTA, Pedro Alberto; BORNIA, Antonio Cezar; REIS, Marcelo Menezes. **Estatística para cursos de engenharia e informática**. 3. ed. São Paulo: Atlas, 2010.

HORST, Allison Marie; HILL, Alison Presmanes; GORMAN, Kristen B. **palmerpenguins: Palmer Archipelago (Antarctica) penguin data**. 2020. DOI: 10.5281/zenodo.3960218.

MONTGOMERY, Douglas C.; RUNGER, George C. **Applied Statistics and Probability for Engineers**. 7. ed. Hoboken: Wiley, 2018.

MORDEN, Peter. **Quantitative Research Methods for the Applied Human Sciences**. 2024.

NAVIDI, William. **Statistics for Engineers and Scientists**. 6. ed. New York: McGraw Hill, 2024.

PANDAS DEVELOPMENT TEAM. **pandas documentation: DataFrame.sample**. 2026. Disponível em: <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sample.html>.

PINHEIRO, João Ismael Dantas et al. **Estatística básica: a arte de trabalhar com dados**. Rio de Janeiro: Elsevier, 2009.